agent/tcs/client/timer.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2014-2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package tcsclient
+
+import "time"
+
+// timerCallback defines the function pointer type for the callback.
+type timerCallback func(interface{}) error
+
+// timer invokes a callback function at specified intervals.
+type timer struct {
+	active     bool
```

Comment:
I'm wondering (like @euank) if it might be easier to use `time.Ticker`.  If we do intend to keep this closer to how it is right now, it seems like we can have a race condition with this `active` field, since it's read and written multiple places that could potentially interleave.